plugins/modules/aci_esg.py

```diff
@@ -0,0 +1,520 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Sabari Jaganathan (@sajagana) <sajagana@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1', 'status': ['preview'], 'supported_by': 'certified'}
+
+
+DOCUMENTATION = r'''
+---
+module: aci_esg
+short_description: Manage Endpoint Security Groups (ESGs) objects (fv:ESg)
+description:
+- Manage Endpoint Security Groups (ESGs) on Cisco ACI fabrics.
+
+options:
+  tenant:
+    description:
+    - Name of an existing tenant.
+    type: str
+    aliases: [ tenant_name ]
+  ap:
+    description:
+    - Name of an existing application network profile, that will contain the ESGs.
+    type: str
+    aliases: [ app_profile, app_profile_name ]
+  esg:
+    description:
+    - Name of the endpoint security group.
+    type: str
+    aliases: [ esg_name, name ]
+  description:
+    description:
+    - Endpoint security group description.
+    type: str
+    aliases: [ descr ]
+  epg:
+    description:
+    - Existing endpoint group name should be in the same tenant.
+    type: str
+    aliases: [ epg_name ]
+  admin_state:
+    description:
+    - Use C(false) to set Admin Up on the ESG Admin state
+    - Use C(true) to set Admin Shut on the ESG Admin state
+    type: str
+    choices: [ false, true ]
+    default: false
+  vrf:
+    description:
+    - Name of the VRF
+    type: str
+    aliases: [ esg_vrf_name ]
+  intra_esg_isolation:
+    description:
+    - The default value of Intra EsG Isolation is C(unenforced).
+    type: str
+    choices: [ enforced, unenforced ]
+    default: unenforced
+  preferred_group_member:
+    description:
+    - The default value of Preferred Group Member is C(exclude).
+    type: str
+    choices: [ exclude, include ]
+    default: exclude
+  epg_selector_description:
+    description:
+    - Description of the EPG selector option.
+    type: str
+  match_key:
+    description:
+    - ESG Tag Selector Key Name
+    type: str
+  value_operator:
+    description:
+    - The default value of ESG Tag Selector operator type is C(equals).
+    - C(contains) used to match values partially.
+    - C(regex) allows to pass patterns values.
+    type: str
+    choices: [ contains, equals, regex ]
+    default: equals
+  match_value:
+    description:
+    - Value filed of the ESG Tag Selector.
+    type: str
+  tag_selector_description:
+    description:
+    - Description of the ESG Tag Selector.
+    type: str
+  subnet_ip:
+    description:
+    - ESG Subnet IP
+    type: str
+    aliases: [ esg_subnet_ip ]
+  subnet_selector_description:
+    description:
+    - Description of the ESG Subnet Selector option.
+    type: str
+  esg_contract_master:
+    description:
+    - Existing ESG name under the same Application profile
+    type: str
+    aliases: [ esg_contract_master ]
```

Comment:
Selectors and contract masters can have more than one element. They should be their own module.